### ISSUE TITLE
Possible false-positive with memo default export

### ISSUE THREAD

**Issue by SukkaW** [ArnaudBarre/eslint-plugin-react-refresh#44]
See also #27.

Example code:

```tsx
import { memo } from 'react';

const Component = () => {
  return <div />;
};

console.log(Component.name);
// >> "Component"
// So actually `Component` is also named

export default memo(Component);
```

```js
// eslint.config.js
import eslint from '@eslint/js';
import reactRefresh from "eslint-plugin-react-refresh";
import tseslint from "typescript-eslint"

export default tseslint.config(
  eslint.configs.recommended,
  {
    files: ['src/**/*.tsx'],
    // in main config for TSX/JSX source files
    plugins: {
      "react-refresh": reactRefresh,
    },
    rules: {
      "react-refresh/only-export-components": "warn",
    },
  },
  ...tseslint.configs.recommended,
);
```

And [minimum reproduction on ESLint REPL](https://eslint-online-playground.netlify.app/#eNqFkjFvwjAQhf/KyUsANckOBVVqp46lA1LdIUou1DS2I9uhIJT/3ottClTQLo6Se/flvTsfmDVljrtCtg1mzu7YlAnZauPgABKlhh5qoyUkBovSJTOuuCq1sg4eNekUKgdzGI1hvoADVwAGXWcU3FdiC/mC9L3vwZ2HVlgXXeM8evRDGJOE3TG0jVAuI3wt1tnGnryESnTyEN7yjR3sRIW394K1QfsRdDzy0rbp1kKlXkGnl3B2anX2HM+Z27doSyNal4YCZ1cSHLui3dGQ/eKLzQyWWkpUFVZ3Q9nPB6AWDdopvCXD6CeTfDLMPXn3EoA8B6FAFnQEDtTawOtylT8vV2B1Z0oMiKAP6YgX6UAJfkWdXkwn/qePT9N5N7e6c62afRqypxQnLMwOUM6+CqM4uwCGM8so05/TGM9o4W1RfhZrpFVrRcv2HsL8A1/qitzFP3BW4fYJ24GgSkH5SRNt/7PrgdUUDi2tMro9dtwonu7YDcGVS3Ku86Ogu8/6b0WtJzM=).

**Comment by ArnaudBarre** [comment]
Yeah this pattern should be allowed to

**Comment by SukkaW** [comment]
@ArnaudBarre I've created a PR #45 to fix this issue.

**Comment by sparrowek** [comment]
A minor problem is that the PR does not handle React.memo() only memo() correctly

**Comment by ArnaudBarre** [comment]
ah yeah I never use this pattern, I still hope that one day React will be ESM 😂
but should be supported anyway, I'll do a quick fix

**Comment by ptmkenny** [comment]
When will this fix be available? I just tried out this plugin today and was stumped why I was getting errors with

`export default React.memo(MyComponent);`


> Fast refresh only works when a file only exports components. Move your component(s) to a separate file.
> Fast refresh can't handle anonymous components. Add a name to your export.

**Comment by ArnaudBarre** [comment]
Sorry I though I'll do it directly and forgot about it! Available in 0.4.10 thanks to @SukkaW!

**Comment by ptmkenny** [comment]
Thanks to both of you!
